src/braket/circuits/diagram_builders/braket_circuit_drawer.py

```diff
@@ -0,0 +1,402 @@
+        elif hasattr(instr, "targets"):
+            qubits = list(instr.targets)
+
+        # Keep track of the highest qubit index
+        if qubits:
+            max_qubit_index = max(max_qubit_index, max(qubits))
+
+        instructions.append({
+            "gate": gate_name,
+            "qubits": qubits,
+            "col_index": idx,
+            "instr": instr,
+        })
+
+    n_qubits = (max_qubit_index + 1) if max_qubit_index >= 0 else 0
+    return instructions, n_qubits
+
+
+def draw_circuit_matplotlib(circuit, figsize=(8, 6), dx=1.5, dy=1.0):
+    """
+    Draw a Braket Circuit using Matplotlib.
+
+    Parameters:
+      - circuit: braket.circuits.Circuit object
+      - figsize: tuple (width, height) in inches. If any is None, computes size automatically.
+      - dx: horizontal spacing between columns
+      - dy: vertical spacing between qubit wires
+
+    Returns:
+      - fig, ax: Matplotlib Figure and Axes instances
+    """
+    instructions, n_qubits = parse_circuit(circuit)
+    if n_qubits == 0:
+        raise ValueError("The circuit contains no qubits to visualize.")
+
+    # Compute figure size if not explicitly provided
+    if figsize[0] is None or figsize[1] is None:
+        width = (len(instructions) + 2) * dx
+        height = (n_qubits + 2) * dy
+        fig, ax = plt.subplots(figsize=(width, height))
+    else:
+        fig, ax = plt.subplots(figsize=figsize)
+
+    ax.set_aspect("equal")
+    ax.axis("off")
+
+    def y_of(q):
+        # Map qubit index q to a y-coordinate: q=0 → y=0, q=1 → y=-dy, etc.
+        return -q * dy
+
+    # Draw horizontal “wires” for each qubit
+    x_min = -dx
+    x_max = (len(instructions) + 1) * dx
+    for q in range(n_qubits):
+        y = y_of(q)
+        ax.hlines(y, x_min, x_max, color="black", linewidth=1.0)
+
+    # Helper to draw a measurement “meter” symbol at (x_ctr, yq)
+    def _draw_meter(x_ctr, yq, size=0.15 * dy):
+        # A circle with no fill and a “⎲”-like tick inside
+        circle = Circle((x_ctr, yq), size, fill=False, edgecolor="black", linewidth=1.0)
+        ax.add_patch(circle)
+        # Draw a small “T” shape inside to indicate a meter
+        # Horizontal cap
+        ax.hlines(yq + size * 0.3, x_ctr - size * 0.5, x_ctr + size * 0.5, color="black", linewidth=1.0)
+        # Vertical stem
+        ax.vlines(x_ctr, yq - size * 0.3, yq + size * 0.3, color="black", linewidth=1.0)
+
+    # Iterate over instructions to draw gates
+    for instr_dict in instructions:
+        gate = instr_dict["gate"]
+        qubits = instr_dict["qubits"]
+        col = instr_dict["col_index"]
+        instr = instr_dict["instr"]
+        x_ctr = col * dx
+
+        if not qubits:
+            continue  # ignore instructions with no qubits
+
+        # 1) Single-qubit parameterized gates (RX, RY, RZ, PHASE, U3, etc.)
+        if len(qubits) == 1 and gate in ["RX", "RY", "RZ", "PHASE", "U3"]:
```

Comment:
We shouldn't hard-code the list of supported gates here. This should support any single-qubit gate (even a custom one), and detect from the gate how many parameters it has.

Reply:
Similar comment for the other gate types below.